spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/ChannelBindingServiceProperties.java

```diff
@@ -234,15 +166,71 @@ public String getBinder(String channelName) {
 		properties.put("instanceCount", String.valueOf(getInstanceCount()));
 		properties.put("defaultBinder", getDefaultBinder());
 		properties.put("dynamicDestinations", getDynamicDestinations());
-		// Add Bindings properties
-		for (Map.Entry<String, BindingProperties> entry : getBindings().entrySet()) {
+		for (Map.Entry<String, BindingProperties> entry : bindings.entrySet()) {
 			properties.put(entry.getKey(), entry.getValue().toString());
 		}
-		// Add Binder config properties
 		for (Map.Entry<String, BinderProperties> entry : binders.entrySet()) {
 			properties.put(entry.getKey(), entry.getValue());
 		}
 		return properties;
 	}
 
+	public <T extends ConsumerProperties> T getConsumerProperties(String inputChannelName, Class<T> beanClass) {
+		Assert.notNull(inputChannelName, "The input channel name cannot be null");
+		Assert.notNull(beanClass, "The bean class cannot be null");
+		T consumerProperties = populateProperties(inputChannelName, beanClass, consumerDefaults);
+		consumerProperties.setCount(this.instanceCount);
+		consumerProperties.setInstanceIndex(this.instanceIndex);
+		return consumerProperties;
+	}
+
+
+	public <T extends ProducerProperties> T getProducerProperties(String outputChannelName, Class<T> beanClass) {
+		Assert.notNull(outputChannelName, "The output channel name cannot be null");
+		Assert.notNull(beanClass, "The bean class cannot be null");
+		T producerProperties = populateProperties(outputChannelName, beanClass, producerDefaults);
+		return producerProperties;
+	}
+
+
+	private <C> C populateProperties(String channelName, Class<C> propertiesClass, Properties defaults) {
+		C beanInstance;
+		try {
+			beanInstance = propertiesClass.newInstance();
+		}
+		catch (InstantiationException | IllegalAccessException e) {
+			throw new BeanInitializationException(e.getMessage());
+		}
+		// bind defaults first
+		RelaxedDataBinder dataBinder = new RelaxedDataBinder(beanInstance);
+		dataBinder.setIgnoreUnknownFields(false);
+		dataBinder.setDisallowedFields(bindingPropertyFields);
+		dataBinder.bind(new MutablePropertyValues(defaults));
+		// bind configured properties next, if available
+		dataBinder = new RelaxedDataBinder(beanInstance, "spring.cloud.stream.bindings." + channelName);
```

Comment:
Can we move this prefix `spring.cloud.stream.bindings` to a constant? and probably use a constant for `spring.cloud.stream` prefix in `ChannelBindingServiceProperties` itself?